### ISSUE TITLE
Errors from ready check are emitted to client handler instead of being bubbled up to caller.

### ISSUE THREAD

**Issue by 3choBoomer** [NodeRedis/node_redis#1093]
* **Version**: 2.6.2
* **Platform**: Node 0.12.2 / Windows Server 2012 R2
* **Description**: If an error is thrown in ready check, it is emitted through the client's event handler instead of being bubbled up to the calling method.

Example:

1. Server is busy running a script. 
2. ready check's call for info returns: `ReplyError: BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.`
3. error is emitted via:
```
err.message = 'Ready check failed: ' + err.message;
this.emit('error', err);
```
4. calling method has no chance to handle the error.

Example of how it works with ready check:
```
var redis = Promise.promisifyAll(require('redis'));
var client = redis.createClient(redisConfig.port, redisConfig.host);

client.on('error', function(err){ //BUSY error ends up here });

function handleError(err) { //BUSY error NEVER gets here };

client.evalAsync(cmd).then(handleReply).error(handleError);
```

Example of how it works with `no_ready_check: true`:
```
var redis = Promise.promisifyAll(require('redis'));
var client = redis.createClient(redisConfig.port, redisConfig.host, {no_ready_check: true});

client.on('error', function(err){ //BUSY error NEVER ends up here });

function handleError(err) { //BUSY error ends up here };

client.evalAsync(cmd).then(handleReply).error(handleError);
```

Shouldn't the ready check either:
1. Handle BUSY responses from the server, retrying just like it does when the server is loading data from disk.
OR
2. Bubble the error up to the caller instead of through the client.

**Comment by BridgeAR** [comment]
This is not how this works. The eval command is never fired when you use the ready check and the server is not yet ready. Therefore the error is returned in the ready check that does not correspond to any specific command that could be rejected. In fact, there should neither be a error or a result for the `eval` command.

Nevertheless I see the point in checking for being busy while connecting.

**Comment by 3choBoomer** [comment]
So the ready check happens only when you create the client, and never again?

Thanks for taking a look at this. 

I think that the retrying on BUSY in the ready check will definitely increase the stability of the module!

**Comment by mivanovaxway** [comment]
@3choBoomer nope, the RedisClient.prototype.ready_check also seems to be called on reconnect.
